src/routers/router.ts

```diff
@@ -56,3 +56,13 @@ export abstract class IRouter<RoutingConfig> {
     routingConfig?: RoutingConfig
   ): Promise<SwapRoute<TradeType.EXACT_OUTPUT> | null>;
 }
+
+export abstract class ISwapToRatio<RoutingConfig> {
```

Comment:
Best if this is an interface, then AlphaRouter can implement it too (in addition to IRouter) so we'd have something like `class AlphaRouter implements IRouter, ISwapToRatio { }`

Also I think you forgot to update AlphaRouter so this is currently unused anyway